DesignModule04/config_man/src/main.rs

```diff
@@ -67,33 +65,29 @@ fn parse_config(config_str: &str) -> HashMap<String, String> {
 }
 
 fn format_config(config: &HashMap<String, String>) -> String {
-    config
-        .iter()
-        .map(|(k, v)| format!("{} = {}", k, v))
+    let mut keys: Vec<&String> = config.keys().collect();
+    keys.sort();
+    keys.iter()
+        .map(|k| format!("{} = {}", k, config.get(*k).unwrap()))
```

Comment:
Can I know why using a reference here ? Doesn't it make an extra memory allocation ?